airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -111,4 +125,105 @@ private WbConnectionRead buildWbConnectionRead(ConnectionRead connectionRead) th
+    Map<String, SourceSchemaStream> originalStreamsByName = original.getStreams().stream()
+        .collect(toMap(SourceSchemaStream::getName, s -> s));
+
+    List<SourceSchemaStream> streams = new ArrayList<>();
+
+    for (SourceSchemaStream stream : discovered.getStreams()) {
+      SourceSchemaStream originalStream = originalStreamsByName.get(stream.getName());
+
+      if (originalStream != null) {
+        Set<String> fieldNames = stream.getFields().stream().map(SourceSchemaField::getName).collect(toSet());
+        stream.setSelected(originalStream.getSelected());
+
+        if (stream.getSupportedSyncModes().contains(originalStream.getSyncMode())) {
+          stream.setSyncMode(originalStream.getSyncMode());
+        }
+
+        Set<String> updatedCursorFields = new HashSet<>();
+
+        for (String oldCursorField : originalStream.getCursorField()) {
+          if (fieldNames.contains(oldCursorField)) {
+            updatedCursorFields.add(oldCursorField);
+          }
+        }
+
+        stream.setCursorField(new ArrayList<>(updatedCursorFields));
+
+        Map<String, SourceSchemaField> originalFieldsByName = originalStream.getFields().stream()
+            .collect(toMap(SourceSchemaField::getName, f -> f));
+
+        for (SourceSchemaField field : stream.getFields()) {
+          if (originalFieldsByName.containsKey(field.getName())) {
+            SourceSchemaField originalField = originalFieldsByName.get(field.getName());
+            field.setSelected(originalField.getSelected());
+          }
+        }
+
+        streams.add(stream);
+      }
+
+    }
+
+    return new SourceSchema().streams(streams);
+  }
+
+  public ConnectionRead webBackendUpdateConnection(WebBackendConnectionUpdate webBackendConnectionUpdate)
+      throws ConfigNotFoundException, IOException, JsonValidationException {
+    final ConnectionUpdate connectionUpdate = extractConnectionUpdate(webBackendConnectionUpdate);
+    final ConnectionRead connectionRead = connectionsHandler.updateConnection(connectionUpdate);
+
+    if (webBackendConnectionUpdate.getWithRefreshedCatalog() != null && webBackendConnectionUpdate.getWithRefreshedCatalog()) {
+      ConnectionIdRequestBody connectionId = new ConnectionIdRequestBody().connectionId(webBackendConnectionUpdate.getConnectionId());
+
+      // wait for this to execute
+      JobInfoRead resetJob = schedulerHandler.resetConnection(connectionId);
+
+      if (!resetJob.getJob().getStatus().equals(JobStatus.SUCCEEDED)) {
+        throw new RuntimeException("Resetting data after updating the connection failed! Please manually reset your data and launch a manual sync.");
+      }
+
+      // just create the job
+      schedulerHandler.syncConnection(connectionId, false);
+    }
+
+    return connectionRead;
+  }
+
+  @VisibleForTesting
+  protected static ConnectionUpdate extractConnectionUpdate(WebBackendConnectionUpdate webBackendConnectionUpdate) {
```

Comment:
`extractConnectionUpdate ` => `toConnectUpdate`? slightly more consistent with existing naming. but if you prefer as is nbd.